packages/components/src/range-control/utils.js

```diff
@@ -2,69 +2,48 @@
 /**
  * External dependencies
  */
-import { clamp, noop } from 'lodash';
+import { noop } from 'lodash';
 
 /**
  * WordPress dependencies
  */
 import { useCallback, useRef, useEffect, useState } from '@wordpress/element';
 
 /**
- * Internal dependencies
- */
-import { useControlledState } from '../utils/hooks';
-
-/**
- * A float supported clamp function for a specific value.
- *
- * @param {number|null} value The value to clamp.
- * @param {number}      min   The minimum value.
- * @param {number}      max   The maximum value.
- *
- * @return {number} A (float) number
- */
-export function floatClamp( value, min, max ) {
-	if ( typeof value !== 'number' ) {
-		return null;
-	}
-
-	return parseFloat( clamp( value, min, max ) );
-}
-
-/**
- * Hook to store a clamped value, derived from props.
+ * Enables entry of out-of-range and invalid values that diverge from state.
  *
- * @param {Object} settings         Hook settings.
- * @param {number} settings.min     The minimum value.
- * @param {number} settings.max     The maximum value.
- * @param {number} settings.value   The current value.
- * @param {any}    settings.initial The initial value.
+ * @param {Object}                 props          Props
+ * @param {number|null}            props.value    Incoming value.
+ * @param {number}                 props.max      Maximum valid value.
+ * @param {number}                 props.min      Minimum valid value.
+ * @param {(next: number) => void} props.onChange Callback for changes.
  *
- * @return {[*, Function]} The controlled value and the value setter.
+ * @return {Object} Assorted props for the input.
  */
-export function useControlledRangeValue( {
-	min,
-	max,
-	value: valueProp,
-	initial,
-} ) {
-	const [ state, setInternalState ] = useControlledState(
-		floatClamp( valueProp, min, max ),
-		{ initial, fallback: null }
-	);
-
-	const setState = useCallback(
-		( nextValue ) => {
-			if ( nextValue === null ) {
-				setInternalState( null );
-			} else {
-				setInternalState( floatClamp( nextValue, min, max ) );
-			}
-		},
-		[ min, max ]
-	);
-
-	return [ state, setState ];
+export function useUnimpededRangedNumberEntry( { max, min, onChange, value } ) {
+	const ref = useRef();
+	const isDiverging = useRef( false );
+	/** @type {import('../input-control/types').InputChangeCallback}*/
+	const changeHandler = ( next ) => {
+		next = parseFloat( next );
+		const isOverflow = next < min || next > max;
+		isDiverging.current = isNaN( next ) || isOverflow;
+		if ( isOverflow ) {
+			next = Math.max( min, Math.min( max, next ) );
+		}
+		onChange( next );
+	};
+	useEffect( () => {
+		if ( ref.current && isDiverging.current ) {
+			const input = ref.current;
+			const entry = input.value;
+			const { defaultView } = input.ownerDocument;
+			defaultView.requestAnimationFrame( () => ( input.value = entry ) );
+			isDiverging.current = false;
+		}
+	}, [ value ] );
```

Comment:
I added a comment in 085fd03 and then while double checking things while testing out the change that was to be 2b14b0f, I discovered the hook didn't need to handle `NaN` values so I made 29634d7. The issue (as non-critical as could be) was a lag before re-syncing the input:

https://user-images.githubusercontent.com/9000376/167703859-42878b30-ec8d-4dd1-907d-ea92d288f3fa.mp4

I'd missed this before https://github.com/WordPress/gutenberg/pull/40568#issuecomment-1115271440 because I dragged the slider instead of clicking it once.